web/packages/teleport/src/Discover/AwsMangementConsole/SetupAccess/SetupAccess.tsx

```diff
@@ -0,0 +1,179 @@
+/**
+ * Teleport
+ * Copyright (C) 2024  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React, { useState, useEffect } from 'react';
+import { LabelInput, Text, Link } from 'design';
+import { useTheme } from 'styled-components';
+import { Cross } from 'design/Icon';
+
+import { OutlineInfo } from 'design/Alert/Alert';
+import { FieldSelectCreatable } from 'shared/components/FieldSelect';
+import Validation, { Validator } from 'shared/components/Validation';
+
+import { Option } from 'teleport/Discover/Shared/SelectCreatable';
+import { IAM_ROLE_ARN_REGEX } from 'teleport/services/integrations/aws';
+import {
+  useUserTraits,
+  SetupAccessWrapper,
+} from 'teleport/Discover/Shared/SetupAccess';
+import { Mark } from 'teleport/Discover/Shared';
+import { styles } from 'teleport/Discover/Shared/SelectCreatable/SelectCreatable';
+
+export function SetupAccess() {
+  const {
+    onProceed,
+    initSelectedOptions,
+    getFixedOptions,
+    getSelectableOptions,
+    ...restOfProps
+  } = useUserTraits();
+
+  const theme = useTheme();
+
+  const [selectedArns, setSelectedArns] = useState<Option[]>([]);
+  const [inputValue, setInputValue] = useState('');
+
+  useEffect(() => {
+    if (restOfProps.attempt.status === 'success') {
+      setSelectedArns(initSelectedOptions('awsRoleArns'));
+    }
+  }, [restOfProps.attempt.status]);
+
+  function handleKeyDown(event: React.KeyboardEvent, validator: Validator) {
+    switch (event.key) {
+      case 'Enter':
+      case 'Tab':
+        if (!validator.validate()) return;
+        setSelectedArns([
```

Comment:
Let's use an updater function, since the new state depends on the old:
```ts
setSelectedArns(prevArns => ([...prevArns, { value: inputValue, label: inputValue }])
```